src/serialize/serializer.rs

```diff
@@ -577,6 +579,12 @@ impl PyObject {
     {
         let ob_type = ob_type!(self.ptr);
 
+        if py_is!(ob_type, BYTEARRAY_TYPE) {
+            return ByteArray::new(self.ptr).serialize(serializer);
+        } else if py_is!(ob_type, MEMORYVIEW_TYPE) {
```

Comment:
nit: remove the else and handle them last, in both `serialize_unlikely` functions.